Chart: Move `CHANGELOG.md` to `changelog/Changelog-4.4.0-and-before.md`.

Gacko opened the pull request:
<!--- Provide a general summary of your changes in the Title above --->
<!--- Please don't @-mention people in PR or commit messages (do so in an additional comment). --->
<!--- Please make sure you title is descriptive, it is used in the Release notes to let others know what it does ---> 

## What this PR does / why we need it:
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

Moves the old `CHANGELOG.md` the `changelog/` directory so we're having all the changelog entries of the past in the same place, which makes it easier and more intuitive to find them.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] CVE Report (Scanner found CVE and adding report)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation only

## Which issue/s this PR fixes
<!--
(optional, in `fixes #<issue number>` format, will close that issue when PR gets merged):

fixes #
-->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/kubernetes/ingress-nginx/blob/main/CONTRIBUTING.md) guide
- [ ] I have added unit and/or e2e tests to cover my changes.
- [ ] All new and existing tests passed.

Gacko commented:
Maybe knowing how I created the single `Changelog-*.md` files makes reviewing a bit easier:

```zsh
previous=TODO
for version in $(grep "###" CHANGELOG.md | awk '{print $2}' | grep --invert-match TODO | sort --version-sort)
do
  cp changelog-template.md changelog/Changelog-${version}.md
  grep "### ${version}" --after-context=1000 CHANGELOG.md | grep --before-context=1000 "### ${previous}" | grep --invert-match "### ${previous}" >> changelog/Changelog-${version}.md
  echo "**Full Changelog**: https://github.com/kubernetes/ingress-nginx/compare/helm-chart-${previous}...helm-chart-${version}" >> changelog/Changelog-${version}.md
  previous="${version}"
done
```

rikatz commented:
Nice! can we just name those as Changelog-chart-version?

The reason is that, by our own fault we have two kinds of versioning, one for controllers and one for charts and I'm afraid this generates some confusion if, at some point of our existance we reach ingress-controller version 4

Gacko commented:
Hm, so you want all of them to be named as `Changelog-helm-chart-4.8.3.md` e.g.? I guess we would also need to change the Go code producing them then.

Apart from that I feel like it's still distinguished enough as those documents are lying in the chart's folder and the controller related changelogs are put elsewhere:

https://github.com/kubernetes/ingress-nginx/blob/da51393cac818545de796d3c3486c40127aa51f4/magefiles/utils/releasenote.go#L47

But yeah, if it's fine for you, I can also move them to something like `changelog-helm-chart-4.8.3.md` and change this in the Go code. I'd also make them start with a lowercase C then.

rikatz commented:
Tbh I don't decide on this :P so I will defer to @strongjz

cpanato commented:
This is for the charts?

Gacko commented:
It was meant to be, but due to @rikatz comment I figured it might be best to also do it for the controller.

So wdyt about just moving both the changelogs to `changelog/` (in the root directory) and name them like `changelog/controller-x.x.x.md` and `changelog/helm-chart-x.x.x` - so just like their tags and release names.

If you do not want them to be in the same location and rather keep the chart's changelog in `charts/ingress-nginx/changelog`, I could still do the renaming for those locations.

Gacko commented:
I now renamed and reworked the changelogs for both, the controller and its chart. See the commits for details. I tried to make them as atomic as possible.

Gacko commented:
Found a minor issue in the newly generated single chart changelogs caused by how I was `grep`ing. Fixed, rebased and pushed. Still every step is split into a separate commit, so it should be fairly easy to review.

I did not touch the `Changelog.md` in the root since it might be referenced by the interwebs and its format changed a lot in the past, so an automatic splitting into separate changelog files would be hard to impossible.

strongjz commented:
/lgtm

let's wait for 1.10 to make this change.

Gacko commented:
/cherry-pick release-1.9
